Zend/zend_object_handlers.c

```diff
@@ -1707,6 +1707,9 @@ ZEND_API zend_function *zend_get_property_hook_trampoline(
 		func = (zend_function *)(uintptr_t)ecalloc(1, sizeof(zend_internal_function));
 	}
 	func->type = ZEND_INTERNAL_FUNCTION;
+	/* This trampoline does not use the call_trampoline_op, so it won't reuse the call frame,
+	 * which means we don't even need to reserve a temporary for observers. */
```

Comment:
I'm a bit confused by this comment. Won't this simply not use observers because of `ZEND_ACC_CALL_VIA_TRAMPOLINE`? See `zend_observer_handler_is_unobserved()`.

Reply:
To clarify, it's not because a function isn't observed that the call frame won't be reused by a function that becomes observed. The reuse happens when dealing with `call_trampoline_op`. This doesn't happen here so we don't have to account for the extra observer temporary.

Reply:
Thanks. This goes over my head at this time of day ^^ I will have a closer look tomorrow.